delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/indexing/CompositeIndexingStreamEntry.scala

```diff
@@ -99,21 +103,32 @@ final case class CompositeIndexingStreamEntry(
       resource: IndexingData,
       includeMetadata: Boolean,
       sourceAsText: Boolean,
-      context: ContextValue
+      context: ContextValue,
+      includeContext: Boolean
   ): Task[Json] = {
     val predGraph = resource.graph
     val metaGraph = resource.metadataGraph
     val graph     = if (includeMetadata) predGraph ++ metaGraph else predGraph
+
     if (sourceAsText)
       graph
         .add(nxv.originalSource.iri, resource.source.noSpaces)
         .toCompactedJsonLd(context)
         .map(_.obj.asJson)
-    else if (resource.source.isEmpty())
-      graph
+    else if (resource.source.isEmpty()) {
+      if (!includeContext) graph
         .toCompactedJsonLd(context)
         .map(_.obj.asJson)
-    else
+      else {
+       val compacted = graph
+          .toCompactedJsonLd(context)
+
+        // TODO: do not exclude context object from document
+        //val ctx = compacted.map(_.ctx.asJson)
```

Comment:
@bogdanromanx Here, I'd like to combine/merge and return the ctx and obj that were serialised to JSON before. I am not familiar with `circe`, I'd be glad for any hint.

My first approach was to serialise the whole `CompactedJsonLd` to JSON but I think that is not the way to go.

Reply:
> could you please confirm your intention? As far as I understand/remember, you'd like to retain the @context value in the resulting elasticsearch document, is that correct?

@bogdanromanx Thanks for your reaction.

Yes, that is exactly my intention. When I query the ES projection, I would like to get back the indexed document *including* the JSON-LD context. For performance reasons, we sometimes prefer to do ES queries. However, without the context the results are not self-contained (not directly processable in the RDF sense).

Reply:
I included the proposed changes in https://github.com/BlueBrain/nexus/pull/3245/commits/6bbfb42bdaedaab98836b84fe62456aa8fb158ca

Please not that I added `asJson` inside `handleCtxInclusion`: `.deepMerge(ctx.contextObj.asJson)`

Reply:
Now,the spec fails since I set the new `includeContext` to `true`: https://github.com/BlueBrain/nexus/blob/6bbfb42bdaedaab98836b84fe62456aa8fb158ca/delta/plugins/composite-views/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/indexing/CompositeIndexingSpec.scala#L195

> [info] - should index resources *** FAILED ***
[info]   The code passed to eventually never returned normally. Attempted 37 times over 30.654083506000003 seconds. Last failure message: forAll failed, because: 
[info]     at index 0, Both Json are not equal (ignoring array order)
[info]     {
[info]       "@context" : {
[info]         "@base" : "http://music.com/",
[info]         "@vocab" : "http://music.com/",
[info]         "_constrainedBy" : {
[info]           "@id" : "https://bluebrain.github.io/nexus/vocabulary/constrainedBy",
[info]           "@type" : "@id"
[info]         },
[info]         "_createdAt" : {
[info]           "@id" : "https://bluebrain.github.io/nexus/vocabulary/createdAt",
[info]           "@type" : "http://www.w3.org/2001/XMLSchema#dateTime"
[info]         },
[info]         "_createdBy" : {
[info]           "@id" : "https://bluebrain.github.io/nexus/vocabulary/createdBy",
[info]           "@type" : "@id"
[info]         },
[info]         "_deprecated" : "https://bluebrain.github.io/nexus/vocabulary/deprecated",
[info]         "_incoming" : {
[info]           "@id" : "https://bluebrain.github.io/nexus/vocabulary/incoming",
[info]           "@type" : "@id"
[info]         },
[info]         "_instant" : "https://bluebrain.github.io/nexus/vocabulary/instant",
[info]         "_outgoing" : {
[info]           "@id" : "https://bluebrain.github.io/nexus/vocabulary/outgoing",
[info]           "@type" : "@id"
[info]         },
[info]         "_project" : {
[info]           "@id" : "https://bluebrain.github.io/nexus/vocabulary/project",
[info]           "@type" : "@id"
[info]         },
[info]         "_rev" : "https://bluebrain.github.io/nexus/vocabulary/rev",
[info]         "_schemaProject" : {
[info]           "@id" : "https://bluebrain.github.io/nexus/vocabulary/schemaProject",
[info]           "@type" : "@id"
[info]         },
[info]         "_self" : {
[info]           "@id" : "https://bluebrain.github.io/nexus/vocabulary/self",
[info]           "@type" : "@id"
[info]         },
[info]         "_source" : "https://bluebrain.github.io/nexus/vocabulary/source",
[info]         "_subject" : "https://bluebrain.github.io/nexus/vocabulary/metadata/subject",
[info]         "_updatedAt" : {
[info]           "@id" : "https://bluebrain.github.io/nexus/vocabulary/updatedAt",
[info]           "@type" : "http://www.w3.org/2001/XMLSchema#dateTime"
[info]         },
[info]         "_updatedBy" : {
[info]           "@id" : "https://bluebrain.github.io/nexus/vocabulary/updatedBy",
[info]           "@type" : "@id"
[info]         },
[info]         "_uuid" : "https://bluebrain.github.io/nexus/vocabulary/uuid",
[info]         "by" : {
[info]           "@type" : "@id"
[info]         }
[info]       },
[info]       "@id" : "muse",
[info]       "album" : {
[info]         "@id" : "absolution",
[info]         "title" : "Absolution"
[info]       },
[info]       "genre" : [
[info]         "electronica",
[info]         "progressive rock",
[info]         "alternative rock"
[info]       ],
[info]       "name" : "Muse",
[info]       "start" : 1994
[info]     }
[info]     did not equal
[info]     {
[info]       "@id" : "muse",
[info]       "album" : {
[info]         "@id" : "absolution",
[info]         "title" : "Absolution"
[info]       },
[info]       "genre" : [
[info]         "electronica",
[info]         "progressive rock",
[info]         "alternative rock"
[info]       ],
[info]       "name" : "Muse",
[info]       "start" : 1994
[info]     } (CompositeIndexingSpec.scala:454)